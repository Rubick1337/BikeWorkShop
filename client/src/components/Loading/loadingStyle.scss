/* Контейнер для выравнивания велосипеда по экрану */
.container__loading {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  position: relative;
  overflow: hidden; /* Чтобы элементы не выходили за пределы экрана */
}

/* Анимация для передвижения велосипеда */
.bike {
  position: absolute;
  left: 0;
  bottom: 47%;
  animation: moveBike 5s linear infinite;
  z-index: 2; /* Убедимся, что велосипед будет над дорогой */
}

/* Анимация для вращения колес */
.wheel {
  position: absolute;
  width: 40px;
  height: 40px;
  background-image: url(../../images/wheelLoading.png);
  background-repeat: no-repeat;
  background-position: center center;
  background-size: contain;
  animation: rotateWheel 1s linear infinite; /* Добавляем анимацию вращения */
}

.front {
  left: 70px;
  bottom: 0;
}

.back {
  left: 0;
  bottom: 0;
}

/* Стили для рамы велосипеда */
.frame {
  width: 140px;
  height: 114px;
  position: absolute;
  left: -19px;
  top: -92px;
  background-image: url(../../images/frameLoading.png);
  background-repeat: no-repeat;
  background-position: center center;
  background-size: contain;
}

/* Дорога */
.road {
  position: absolute;
  left: 0;
  width: 100%;
  height: 200px;
  background-image: url(../../images/asphalt-vector-33.png);
  background-position: center center;
  background-size: contain;
  z-index: 1;
}



/* Анимация движения велосипеда по оси X */
@keyframes moveBike {
  0% {
    transform: translateX(-100px);
  }
  100% {
    transform: translateX(100vw);
  }
}

/* Анимация вращения колес */
@keyframes rotateWheel {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
