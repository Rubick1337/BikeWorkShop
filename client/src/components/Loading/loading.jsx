import React from 'react';
import './loadingStyle.scss';

const Loading = () => {
    return (
        <div className="container__loading">
            <div className="road"></div> {/* Дорога */}
            <div className="bike">
                <div className="wheel front"></div>
                <div className="wheel back"></div>
                <div className="frame"></div>
            </div>
        </div>
    );
};

export default Loading;
