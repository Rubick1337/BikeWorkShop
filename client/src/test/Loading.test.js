import React from 'react';
import { render } from '@testing-library/react';
import Loading from '../components/Loading/Loading'; // Убедитесь, что путь правильный

describe('Loading Component', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(<Loading />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders loading elements correctly', () => {
        const { getByText } = render(<Loading />);

        // Проверяем наличие класса для загрузки
        expect(document.querySelector('.container__loading')).toBeInTheDocument();
        expect(document.querySelector('.bike')).toBeInTheDocument();
        expect(document.querySelector('.wheel')).toBeInTheDocument();
    });
});