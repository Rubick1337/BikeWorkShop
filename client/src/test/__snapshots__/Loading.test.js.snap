// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container__loading"
  >
    <div
      class="road"
    />
     
    <div
      class="bike"
    >
      <div
        class="wheel front"
      />
      <div
        class="wheel back"
      />
      <div
        class="frame"
      />
    </div>
  </div>
</DocumentFragment>
`;
